djtriggers/models.py

```diff
@@ -82,28 +90,95 @@ def set_source(self, *args):
     def get_source(self):
         return tuple(x for x in self.source.split('$') if x != '')
 
-    def process(self, force=False, logger=None, dictionary={}):
-        if logger:
-            self.logger = get_logger(logger)
-        now = timezone.now()
-        if not force and self.date_processed is not None:
-            raise AlreadyProcessedError()
-        if not force and self.process_after and self.process_after >= now:
-            raise ProcessLaterError(self.process_after)
+    @shared_task(filter=task_method)
+    def process(self, force=False, logger=None, dictionary={}, use_statsd=False):
+        """
+        Executes the Trigger
+        :param bool force: force the execution
+        :param string logger: slug of preferred logger
+        :param dict dictionary: dictionary needed by trigger to execute
+        :param bool use_statsd: whether to use statsd
+        :return: None
+        """
+        # The task gets locked because multiple tasks in the queue can process the same trigger.
+        # The lock assures no two tasks can process a trigger simultaneously.
+        # The check for date_processed assures a trigger is not executed multiple times.
+        try:
+            trigger_lock_name = 'execute_trigger%d' % self.pk
+            file_lock = FileLock(trigger_lock_name)
```

Comment:
A FileLock doesn't work when running on multiple servers, so that is not an option. Please have discussion with your fellow Dev's on how to solve that 'non-waiting' problem.